### PR TITLE
Update dependencies and make adjustments for PHP 8.0/8.1

### PR DESCRIPTION
**Pull request by dweichert**
- update psr/log to ^3.0
- replace beberlei/assert with webmozart/assert

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -10,17 +10,16 @@
     }
   ],
   "require": {
-    "php": "~7.2|~7.3|~7.4|~8.1",
-    "psr/log": "^1.0",
-    "beberlei/assert": "^2.4|^3.0",
-    "regex-guard/regex-guard": "^1.1"
+    "php": "8.0.* || 8.1.*",
+    "psr/log": "^3.0",
```
Comment (dkreuer) [inline composer.json]: Is there a chance to make it compatible with all psr/log versions? Just thinking about compatibility with other implementations …

GitHub Actions should provide ne necessary testing mechanisms (locked, lowest, update)

File: src/LogEntry.php [hunk 2]
```diff
@@ -95,7 +95,7 @@ public function __toString()
     private function replaceCurlyBrackets($logMessage, $logContext)
     {
         return preg_replace_callback(
-            '(\{(.*?)\})',
+            '(\{(.*?)})',
```
Comment (dkreuer) [inline src/LogEntry.php:98]: This is probably an error? The opening brace is escaped, the closing one should be too?

File: src/InMemoryLogger.php [hunk 3]
```diff
@@ -26,7 +26,7 @@ public function __construct()
     /**
      * {@inheritdoc}
      */
-    public function log($level, $message, array $context = array())
+    public function log($level, string|\Stringable $message, array $context = []): void
```
Comment (dkreuer) [inline src/InMemoryLogger.php]: `\Stringable` must be imported.

File: composer.json [hunk 4]
```diff
@@ -10,17 +10,16 @@
...
+    "regex-guard/regex-guard": "^1.1",
+    "webmozart/assert": "^1.11"
```
Comment (dkreuer) [inline composer.json:16]: Why switching to `webmozart/assert` since depends graph for `beberlei/assert` looks compatible?

**Review by dkreuer — CHANGES_REQUESTED** [review]

**Comment by dkreuer** [comment]
Would it be better to split the changes into:
1. Switch pipeline implementation from Travis CI to GitHub Actions
2. Updating dependencies

That would allow to ensure the pipline before and after is green.

File: src/LogEntry.php (hunk 2 above)
Comment (dweichert) [reply src/LogEntry.php:98]: PhpStorm inspection shows second escape to be superfluous.

Comment (dkreuer) [reply src/LogEntry.php:98]: Looks like that’s the case (in regex101) .. still not convinced that it helps to understand the regex.

File: composer.json (hunk 4 above)
Comment (dweichert) [reply composer.json:16]: Both libraries generally do the same and are mostly interchangeable. Webmozart has better support for nice messages. I would prefer to use it in all Metasyntactical projects and remove beberlei so that in the future webmozart is the only such dependency in all our libraries.

File: src/LogEntry.php [hunk 5]
```diff
@@ -57,8 +57,8 @@ public function isOfLogLevel(array $logLevelList)
 
     public function containsRegExp($regExp)
     {
-        Assertion::string($regExp);
-        Assertion::minLength($regExp, 1);
+        Assert::string($regExp);
+        Assert::minLength($regExp, 1);
```
Comment (dkreuer) [inline src/LogEntry.php]: Could be of minLength of 2 .. can’t think about a valid regex of length 1 that makes sense in that situation.

File: composer.json (hunk 4 above)
Comment (dkreuer) [reply composer.json:16]: Works for me.

File: composer.json [hunk 6]
```diff
@@ -10,16 +10,16 @@
     }
   ],
   "require": {
-    "php": "8.0.*||8.1.*",
-    "psr/log": "^1.0",
-    "beberlei/assert": "^2.4|^3.0",
-    "regex-guard/regex-guard": "^1.1"
+    "php": "8.0.* || 8.1.*",
```
Comment (dkreuer) [inline composer.json]: Unnecessary whitespace change

Comment (dweichert) [reply composer.json]: Removed change (came automatically when rebasing on master)

File: src/LogEntry.php (hunk 5 above)
Comment (dweichert) [reply src/LogEntry.php]: 👍

File: composer.json (hunk 6 above)
Comment (dkreuer) [reply composer.json]: Weird. Should be the same as on default branch.

**Review by dkreuer — APPROVED** [review]
